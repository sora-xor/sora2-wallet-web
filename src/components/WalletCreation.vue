<template>
  <wallet-base :title="t('creation.title')" show-back @back="handleBack">
    <div class="wallet-creation">
      <s-input :placeholder="t('creation.name.placeholder')" v-model="creationFormData.name" border-radius="mini" />
      <div class="wallet-creation-hint">{{ t('creation.name.hint') }}</div>
      <s-select :placeholder="t('creation.sourceType.placeholder')" :value="creationFormData.sourceType" border-radius="mini">
        <s-option
          :value="SourceTypes.MnemonicSeed"
          :label="t(`creation.sourceType.${SourceTypes.MnemonicSeed}`)"
        />
      </s-select>
      <div class="wallet-creation-hint">{{ t('creation.sourceType.hint') }}</div>
      <s-tooltip
        class="header-tooltip"
        popper-class="info-tooltip info-tooltip--header"
        :content="t('comingSoonText')"
        border-radius="mini"
        theme="light"
        placement="bottom-end"
        animation="none"
        :show-arrow="false"
      >
        <s-button
          type="primary"
          size="medium"
          @click="handleNext"
        >
          {{ t('creation.action') }}
        </s-button>
      </s-tooltip>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action } from 'vuex-class'

import TranslationMixin from './mixins/TranslationMixin'
import WalletBase from './WalletBase.vue'
import { RouteNames, SourceTypes } from '../consts'

@Component({
  components: { WalletBase }
})
export default class WalletCreation extends Mixins(TranslationMixin) {
  readonly SourceTypes = SourceTypes

  @Action navigate

  creationFormData = {
    name: '',
    sourceType: SourceTypes.MnemonicSeed
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletConnection })
  }

  handleNext (): void {
  }
}
</script>

<style lang="scss">
@include select-icon('wallet-creation');
</style>

<style scoped lang="scss">
.wallet-creation {
  &-hint {
    @include hint-text(true);
  }
  > button {
    width: 100%;
  }
  > :not(:last-child) {
    margin-bottom: $basic-spacing;
  }
}
</style>
