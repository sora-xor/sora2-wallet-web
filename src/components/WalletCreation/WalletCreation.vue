<template>
  <wallet-base :title="t('creation.title')" show-back @back="handleBack">
    <div class="wallet-creation">
      <s-input :placeholder="t('creation.name.placeholder')" v-model="creationFormData.name" />
      <div class="wallet-creation-hint">{{ t('creation.name.hint') }}</div>
      <s-select :placeholder="t('creation.sourceType.placeholder')" :value="creationFormData.sourceType">
        <s-option
          :value="SourceTypes.MnemonicSeed"
          :label="t(`creation.sourceType.${SourceTypes.MnemonicSeed}`)"
        />
      </s-select>
      <div class="wallet-creation-hint">{{ t('creation.sourceType.hint') }}</div>
      <s-button
        type="primary"
        size="medium"
        :tooltip="t('comingSoonText')"
        @click="handleNext"
      >
        {{ t('creation.action') }}
      </s-button>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action } from 'vuex-class'

import TranslationMixin from '../mixins/TranslationMixin'
import WalletBase from '../WalletBase.vue'
import { RouteNames, SourceTypes } from '../../consts'

@Component({
  components: { WalletBase }
})
export default class WalletCreation extends Mixins(TranslationMixin) {
  readonly SourceTypes = SourceTypes

  @Action navigate

  creationFormData = {
    name: '',
    sourceType: SourceTypes.MnemonicSeed
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletConnection })
  }

  handleNext (): void {
  }
}
</script>

<style scoped lang="scss">
@import '../../styles/typography';
@import '../../styles/colors';
@import '../../styles/layout';

$font-size-hint: $font-size_small;

.wallet-creation {
  &-hint {
    color: $s-color-content-tertiary;
    font-size: $font-size-hint;
    line-height: 1.8;
  }
  > button {
    width: 100%;
  }
  > :not(:last-child) {
    margin-bottom: $basic-spacing;
  }
}

</style>
