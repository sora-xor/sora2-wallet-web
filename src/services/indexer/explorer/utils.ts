import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
