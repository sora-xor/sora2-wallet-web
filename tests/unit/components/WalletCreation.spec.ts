import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import WalletCreation from '@/components/WalletCreation.vue'
import { TranslationMock, SoramitsuElementsImport } from '../../utils'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  modules: {
    Router: {
      actions: {
        navigate: jest.fn()
      }
    }
  } as any
})

describe('WalletCreation.vue', () => {
  beforeEach(() => {
    SoramitsuElementsImport(localVue)
    TranslationMock(WalletCreation)
  })

  it('should renders correctly', () => {
    const wrapper = shallowMount(WalletCreation, { localVue, store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
