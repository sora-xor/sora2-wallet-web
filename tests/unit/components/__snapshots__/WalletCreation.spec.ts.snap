// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletCreation.vue should renders correctly 1`] = `
<wallet-base-stub
  showback="true"
  title=""
>
  <div
    class="wallet-creation"
  >
    <s-input-stub
      accept=""
      autocomplete="off"
      form=""
      label=""
      name=""
      placeholder=""
      prefix=""
      size="big"
      suffix=""
      tabindex=""
      type="text"
      value=""
    />
     
    <div
      class="wallet-creation-hint"
    >
      
    </div>
     
    <s-select-stub
      autocomplete="off"
      inputtype="input"
      loadingtext="Loading"
      multiplelimit="0"
      multipletextprefix=""
      name=""
      nodatatext="No data"
      placeholder="Select"
      value="MnemonicSeed"
    >
      <s-option-stub
        label=""
        value="MnemonicSeed"
      />
    </s-select-stub>
     
    <div
      class="wallet-creation-hint"
    >
      
    </div>
     
    <s-button-stub
      icon=""
      nativetype="button"
      size="medium"
      tabindex="0"
      tooltip=""
      tooltipplacement="top"
      type="primary"
    >
      
      
    
    </s-button-stub>
  </div>
</wallet-base-stub>
`;
